src/sasctl/_services/folders.py

```diff
@@ -135,3 +137,46 @@ def get_folder(cls, folder, refresh=False):
             return cls.get(f"/folders/{folder}")
 
         return cls._get_folder(folder, refresh=refresh)
+
+    @classmethod
+    def create_path(cls, folder, description=None):
+        """Create a new folder recursively.
+
+        Parameters
+        ----------
+        folder : str
+            The folder to be created including the path.
+        description: str, optional
+             A description of the folder
+
+        Returns
+        -------
+        RestObj
+            Details of newly-created folder
+
+        """
+        if isinstance(folder, str) and folder.count("/") >= 1:
+            # Path must include a leading "/"
+            if not folder.startswith("/"):
+                folder = f"/{folder}"
+        else:
+            raise ValueError(
+                "Invalid folder parameter. The complete path should be provided!")
+
+        path = folder.split("/")
+        level = 1
+        while level < len(path):
+            level += 1
+            current_path = path[0:level]
+            name = current_path[-1]
+            parent = "/".join(current_path[0:-1])
+            if not cls.get_folder("/".join(current_path)):
+                try:
+                    new_folder = cls.create_folder(name, parent=parent, description=description)
+                except HTTPError:
+                    folder_name = "/".join([parent, name])
+                    print(f"Folder {folder_name} could not be created.")
+        try:
```

Comment:
Return value should be the last folder in the path, regardless of whether it was created or already existed.  Can set `new_folder = cls.get_folder(...)` and then check and overwrite with `new_folder = cls.create_folder(...)`.  Alternatively, can be combined into a single statement `new_folder = cls.get_folder(...) or cls.create_folder(...)`.

The results in `new_folder` always being set, so no try/except block needed.

Reply:
I opted for the first option for code readibility.